WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+			echo '  ERROR  | Screenshot width must be 1200px or less.' . PHP_EOL;
+		}
+
+		/**
+		 * ERROR : height must be less then 900 px
+		 */
+		if ( false === $sniff_helper['screenshot']['less_than_900_high'] &&  true === $sniff_helper['screenshot']['found'] ) {
+			echo '  ERROR  | Screenshot height must be 900px or less.' . PHP_EOL;
+		}
+
+		/**
+		 * ERROR : aspect ratio must be 4:3.
+		 */
+		if ( false === $sniff_helper['screenshot']['aspect_ratio_4_by_3'] &&  true === $sniff_helper['screenshot']['found'] ) {
+			echo '  ERROR  | Screenshot aspect ratio must be 4:3.' . PHP_EOL;
+		}
+
+		/**
+		 * WARNING : unable to obtain image details.
+		 */
+		if ( false !== $sniff_helper['screenshot']['details_not_found'] &&  true === $sniff_helper['screenshot']['found'] ) {
+			echo ' WARNING | Image details were not found.' . PHP_EOL;
+			echo '         | Please check to ensure max dimensions of 1200 x 900 px.' . PHP_EOL;
+			echo '         | Aspect ratio must be 4:3.' . PHP_EOL;
+		}
+
+		/**
+		 * ERROR/WARNING : Check on the number of unique text domains encountered.
+		 * If one => ok, if two => warning, if three => error.
+		 */
+		$textdomain_count = count( $sniff_helper['textdomains'] );
+		if ( 2 === $textdomain_count ) {
+			echo ' WARNING | Found 2 text domains :' . PHP_EOL;
+			echo '         | ' . $sniff_helper['textdomains'][0] . PHP_EOL;
+			echo '         | ' . $sniff_helper['textdomains'][1] . PHP_EOL;
+			echo '         | You can have 2 text domains if you are using a framework.' . PHP_EOL;
+			echo '         | If you are not using a framework remove one text domain.' . PHP_EOL;
+			echo '         | Also note that your main text domain must be your WordPress' . PHP_EOL;
+			echo '         | approved themeslug.' . PHP_EOL;
+		} elseif ( $textdomain_count > 2 ) {
+			echo '  ERROR  | Found ' . $textdomain_count . ' text domains :' . PHP_EOL;
+			foreach ( $sniff_helper['textdomains'] as $textdomain ) {
+				echo '         | ' . $textdomain . PHP_EOL;
+			}
+			echo '         | Only one text domain is allowed unless you are using a framework.' . PHP_EOL;
+			echo '         | You can have 2 text domains if you are using a framework.' . PHP_EOL;
+			echo '         | Also note that your main text domain must be your WordPress' . PHP_EOL;
+			echo '         | approved themeslug.' . PHP_EOL;
+		} elseif ( 0 === $textdomain_count ) {
+			echo ' WARNING | No text domain found.' . PHP_EOL;
+			echo '         | Note that all text strings must be translated.' . PHP_EOL;
+		}
+
+		/**
+		 * ERROR | Verify that the text-domain used is the same as the theme slug.
+		 */
+		// core names their themes differently.
+		$textdomain_exceptions = array( 'twentyten', 'twentyeleven', 'twentytwelve', 'twentythirteen', 'twentyfourteen', 'twentyfifteen', 'twentysixteen',  'twentyseventeen', 'twentyeighteen', 'twentynineteen', 'twentytwenty' );
+
+		$textdomain = $sniff_helper['theme_data']['text_domain'];
+		if ( false !== strpos( $textdomain, '_' ) ) {
```

Comment:
The underscore problem was something I though we wanted to check. As the language API has difficulty with this.

I would also suggest that authors who use the _s starter, need to ensure they have made the appropriate changes.

I will leave for now.